policies.go

```diff
@@ -415,16 +427,19 @@ type tokenAwareHostPolicy struct {
 	mu          sync.RWMutex
 	partitioner string
 	fallback    HostSelectionPolicy
-	session     *Session
+	getKeyspaceMetadata func(keyspace string) (*KeyspaceMetadata, error)
+	getKeyspaceName func() string
 
 	tokenRing atomic.Value // *tokenRing
 	keyspaces atomic.Value // *keyspaceMeta
 
-	shuffleReplicas bool
+	shuffleReplicas          bool
+	nonLocalReplicasFallback bool
 }
 
 func (t *tokenAwareHostPolicy) Init(s *Session) {
-	t.session = s
+	t.getKeyspaceMetadata = s.KeyspaceMetadata
```

Comment:
Can the session ever become invalid and if so can this blow up then? I mean the functions borrowed from the live session.

Reply:
> The session isn't needed by something else I presume. I prefer extracting details like this rather than always storing the Session everywhere.

Nope, it's not needing by anything else in tokenAwareHostPolicy, that's why I removed the field altogether.

> Can the session ever become invalid and if so can this blow up then? I mean the functions borrowed from the live session.

Previously the code called the same functions from session and `t.session` was not reassigned anywhere, so I expect the code to behave exactly the same as before.